packages/holocron-dev-server/__tests__/lib/middleware/mocks.spec.js

```diff
@@ -0,0 +1,188 @@
+} from '../../../lib/utils';
+import { error, logMockAction, logScenariosRegistered } from '../../../lib/utils/logs';
+
+jest.mock('parrot-middleware');
+jest.mock('../../../lib/utils/logs');
+jest.mock('../../../lib/utils/helpers', () => ({
+  getModuleFromFilePath: jest.fn((s) => s),
+}));
+jest.mock('../../../lib/utils/virtual-file-system', () => ({
+  ufs: {
+    existsSync: jest.fn(() => true),
+  },
+}));
+jest.mock('../../../lib/utils/publish', () => ({
+  publish: jest.fn(),
+}));
+jest.mock('../../../lib/utils/watcher', () => ({
+  watchFiles: jest.fn(),
+}));
+jest.mock('../../../lib/utils/paths', () => ({
+  joinUrlFragments: jest.fn((...args) => args.join('')),
+  getScenarioPathForModule: jest.fn((s) => `${s}/__mocks__/scenario.js`),
+  getMockDirectoryForModule: jest.fn((s) => `${s}/__mocks__`),
+}));
+
+const mocksDir = path.resolve(__dirname, '../../../__mocks__');
+const modulePath = path.resolve(__dirname, '../../..');
+const scenariosPath = require.resolve('../../../__mocks__/scenario');
+const watcher = {};
+watcher.on = jest.fn(() => watcher);
+watcher.getWatched = jest.fn(() => ({ [mocksDir]: ['scenario.js'] }));
+const { on, getWatched } = watcher;
+beforeAll(() => {
+  parrot.mockImplementation(() => jest.fn());
+  watchFiles.mockImplementation(() => watcher);
+  getScenarioPathForModule.mockImplementation(() => scenariosPath);
+  getMockDirectoryForModule.mockImplementation(() => mocksDir);
+});
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('loadScenarios', () => {
+  test('requires the scenario paths to return result', () => {
+    expect(loadScenarios()).toEqual({});
+    expect(loadScenarios([scenariosPath])).toEqual(require('../../../__mocks__/scenario'));
+  });
+
+  test('loads empty scenario without failing', () => {
+    const scenarioEmptyPath = scenariosPath.replace('scenario', 'empty-scenarios');
+    getScenarioPathForModule.mockImplementationOnce(() => scenarioEmptyPath);
+    expect(loadScenarios([scenarioEmptyPath])).toEqual({});
+  });
+});
+
+describe('getMocksFromModules', () => {
+  test('requires the scenario paths to return result', () => {
+    const modules = [
+      {
+        modulePath,
+      },
+    ];
+    expect(getMocksFromModules()).toEqual([[], []]);
+    expect(getMocksFromModules(modules)).toEqual([[mocksDir], [scenariosPath]]);
+  });
+});
+
+describe('createMockRouter', () => {
+  test('returns an express router with added properties', () => {
+    const router = createMockRouter();
+    expect(router.loadScenarios).toBeInstanceOf(Function);
+    expect(router.reload).toBeInstanceOf(Function);
+    expect(router.reset).toBeInstanceOf(Function);
+    expect(router.get).toBeInstanceOf(Function);
+    expect(router.use).toBeInstanceOf(Function);
+    expect(router.stack).toBeInstanceOf(Array);
+  });
+
+  test('resets the router stack', () => {
+    const router = createMockRouter();
+
+    router.use((_, __, next) => next());
```

Comment:
I don't think we can just have a noop.. we need to call `next`.

> If the current middleware function does not end the request-response cycle, it must call next() to pass control to the next middleware function. Otherwise, the request will be left hanging.

https://expressjs.com/en/guide/using-middleware.html